src/routes/InventoryTable.js

```diff
@@ -177,10 +177,20 @@ const Inventory = ({
   useEffect(() => {
     setActiveTabKey(initialActiveTabKey);
   }, [pathname]);
+  const searchParams = useLocation();
+  const [prm, setPrm] = useState('');
+  const [currentTab, setCurrentTab] = useState(activeTabKey);
   const handleTabClick = (_event, tabIndex) => {
-    const tabPath = tabsPath[tabIndex];
-    if (tabPath !== undefined) {
-      navigate(`${tabPath}`);
+    setCurrentTab(tabIndex);
+    if (currentTab !== tabIndex) {
+      setPrm(searchParams.search);
+      let tabPath = tabsPath[tabIndex];
+      if (tabPath !== undefined) {
+        if (tabPath === '') {
+          tabPath = '/';
+        }
+        navigate(`${tabPath}${prm}`);
+      }
```

Comment:
We might be able to simplify this. We don't need to keep the current tab in a state as well, but only use the searchParams. There is an example in the react router migrate to v6 in remediations[0] as an example and an explanation in Slack[1].


[0] https://github.com/RedHatInsights/insights-remediations-frontend/blob/7400ad209414835b67f637855d5d7ffa8b0a770e/src/routes/RemediationDetails.js#L261-L277
[1] https://redhat-internal.slack.com/archives/C02T90PLJ5V/p1692634189934329

Reply:
@bastilian i've made the change to use the urlSearchParam approach, thanks for the guidance.
@computercamplove the behavior and the ur changed a bit, can you take a look with this new solution?
Thanks

Reply:
![THEEDGE-3474-2](https://github.com/RedHatInsights/insights-inventory-frontend/assets/5039367/f52c5bc5-414f-4ecc-ac69-31c7d20f290e)